internal/appsec/remoteconfig.go

```diff
@@ -61,6 +61,9 @@ func combineRCRulesUpdates(r *rulesManager, updates map[string]remoteconfig.Prod
 updateLoop:
 	// Process rules related updates
 	for p, u := range updates {
+		if u != nil && len(u) == 0 {
+			continue
+		}
```

Comment:
does this happen sometimes? 🤔
if so, isn't `u == nil || len(u) == 0` even broader against that?

Reply:
len(u) == 0 with u != nil means an empty update which is semantically different from u == nil, which means a config removal.
i.e empty update == no change, nil update == remove config.

Reply:
oh yes my bad, totally forgot that one, thanks